sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/aad/AzureJwtBearerTokenAuthenticationConverter.java

```diff
@@ -0,0 +1,48 @@
+package com.azure.spring.autoconfigure.aad;
+
+import com.nimbusds.jose.JWSObject;
+import com.nimbusds.jwt.JWTClaimsSet;
+import com.nimbusds.jwt.JWTClaimsSet.Builder;
+import java.text.ParseException;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Map.Entry;
+import org.springframework.core.convert.converter.Converter;
+import org.springframework.security.authentication.AbstractAuthenticationToken;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.oauth2.core.OAuth2AccessToken;
+import org.springframework.security.oauth2.core.OAuth2AccessToken.TokenType;
+import org.springframework.security.oauth2.jwt.Jwt;
+import org.springframework.security.oauth2.server.resource.authentication.JwtAuthenticationConverter;
+import org.springframework.security.web.authentication.preauth.PreAuthenticatedAuthenticationToken;
+
+/**
+ *  A {@link Converter} that takes a {@link Jwt} and converts it into a {@link PreAuthenticatedAuthenticationToken}.
+ */
+public class AzureJwtBearerTokenAuthenticationConverter implements Converter<Jwt, AbstractAuthenticationToken> {
+
+    private final JwtAuthenticationConverter jwtAuthenticationConverter = new JwtAuthenticationConverter();
+
+    public AzureJwtBearerTokenAuthenticationConverter() {
+    }
+
+    public AbstractAuthenticationToken convert(Jwt jwt) {
+        AbstractAuthenticationToken token = this.jwtAuthenticationConverter.convert(jwt);
+        Collection<GrantedAuthority> authorities = token.getAuthorities();
```

Comment:
Does this includes both scope_ and role_?

Reply:
Here is getting one of the two and seeing which one the user needs.Gets permissions in `scp` by default.Of course, the user can also be configured to obtain permissions in `role`.